File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int maxn=nums[0];
        int maxs=nums[0];
        for(int i=1;i<nums.length;i++){
            maxn=Math.max(maxn+nums[i],nums[i]);
            maxs=Math.max(maxn,maxs);
        }
        return maxs;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxn=nums[0]
        maxs=nums[0]
        minn=nums[0]
        for i in range(1,len(nums)):
            minn=min(nums[i],nums[i]+minn)
            maxn=max(nums[i],nums[i]+maxn)
            maxs=max(maxs,maxn,minn)
        return maxs